pkg/telemetry/events.go

```diff
@@ -47,10 +54,17 @@ func (s *TelemetryService) ParticipantJoined(ctx context.Context, room *livekit.
 	// TODO: analytics service
 }
 
-func (s *TelemetryService) ParticipantLeft(ctx context.Context, room *livekit.Room, participant *livekit.ParticipantInfo) {
-	s.pool.Submit(prometheus.SubParticipant)
+func (t *TelemetryService) ParticipantLeft(ctx context.Context, room *livekit.Room, participant *livekit.ParticipantInfo) {
+	t.Lock()
+	if w := t.workers[participant.Sid]; w != nil {
+		w.Close()
+		delete(t.workers, participant.Sid)
+	}
+	t.Unlock()
+
+	prometheus.SubParticipant()
```

Comment:
Maybe check that a participant was really deleted before calling subtract?

Reply:
it should always be deleted, but it should be easy to spot either way if counts are off